modules/gapi/test/s11n/gapi_s11n_tests.cpp

```diff
@@ -2,6 +2,38 @@
 
 #include "backends/common/serialization.hpp"
 
+namespace {
+    struct MyCustomType {
+        int val;
+        std::string name;
+        cv::Size sz;
+        bool operator==(const MyCustomType& other) const {
+            return val == other.val && name == other.name && sz == other.sz;
+        }
+    };
+}
+
+// FIXME: namespace conflicts
+namespace cv {
+namespace gimpl {
+namespace s11n {
+namespace detail {
+    template<> struct S11N<MyCustomType> {
+        static void serialize(I::OStream &os, const MyCustomType &p) {
+            os << p.val << p.name << p.sz;
+        }
+        static MyCustomType deserialize(I::IStream &is) {
+            MyCustomType p;
+            is >> p.val >> p.name >> p.sz;
+            return p;
+        }
+        static bool isSupported() { return true; }
```

Comment:
I don't like this flag tbh :)

Reply:
I think it's nice to have such an indicator

Reply:
Dropped for now. Can be done separately if will be needed